### PR TITLE
Rel 1.2.4

### PR DESCRIPTION
**Pull request by h00pl4**

### CODE REVIEW

**Comment by h00pl4** [comment]
Noting that, after merge, will need to:
Add to permissions table:
('reports_memberships', 'reports', NULL),
Add to grants table:
('reports_memberships', 1, '--'),

Currently, I've disabled myPHPAdmin on production, as it was a security risk. Should be able to just write a script though to perform through Docker Run.

```sh
docker exec -i cbv-dev-db sh -c 'exec mysql -uroot -p"$MYSQL_ROOT_PASSWORD" $MYSQL_DATABASE' <<< "REPLACE INTO permissions (permission_id, module_id, location_id) VALUES ('reports_memberships', 'reports', NULL); 
REPLACE INTO grants (permission_id, person_id, menu_group) VALUES ('reports_memberships', '1', '--');" 
```

**Comment by rjobeirne** [comment]
I can get this working but it requires a further additional to the database.
Add to 'items' table:
('CBV Membership', 'CBV Membership Fee').

~/app/application/controllers/Items.php is setup for the above entry to be item_id=9. If that's not the case the file will require modification.

Note: I put this comment on the Trello ticket. I now realise I should have put it here.

File: app/application/models/reports/Detailed_memberships.php [hunk 1]
```diff
@@ -0,0 +1,58 @@
...
+                        people.email as email_adr,
+                        sales_items.item_unit_price as amount_paid
+                        ');
+        $this->db->from('sales_items AS sales_items');
+        $this->db->join('sales AS sales', 'sales.sale_id = sales_items.sale_id', 'inner');
+        $this->db->join('people AS people', 'people.person_id = sales.customer_id', 'inner');
+        $this->db->join('customers AS customers', 'customers.person_id = people.person_id', 'inner');
+
+        // Only include Membership item, add
+        $this->db->where('sales_items.item_id = 9');
```
Comment (h00pl4) [inline app/application/models/reports/Detailed_memberships.php:43]: Hmm, I'm not a big fan of this approach, as it's very brittle. Would prefer if the `item_id` was configurable through OSPOS settings or an even an environment variable. For now, will update the SQL dump file that ships with the dev install to have the same "fixed" items as what we have in Prod currently.

**Comment by h00pl4** [comment]
> I can get this working but it requires a further additional to the database. Add to 'items' table: ('CBV Membership', 'CBV Membership Fee').
> 
> ~/app/application/controllers/Items.php is setup for the above entry to be item_id=9. If that's not the case the file will require modification.
> 
> Note: I put this comment on the Trello ticket. I now realise I should have put it here.

Agreed @rjobeirne. I can see in production, that `item_id == 9` is in fact Membership Fees. I spoke about this assumption above. For now, I think we just update the SQL files in the repo to have `item_id == 9` be the Membership Fees item. It's still brittle, but a bit more intentional.

**Review by rjobeirne — APPROVED** [review]
I was surprised about the referencing item_id as well.

**Comment by h00pl4** [comment]
I will attempt to deploy this tomorrow when I'm in @rjobeirne.

**Pull request merged**